presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -250,6 +254,15 @@ public ConnectorSplitSource getSplits(
                 session.getRuntimeStats());
         Table table = layout.getTable(metastore, metastoreContext);
 
+        if (!readFormats.isEmpty()) {
+            StorageFormat storageFormat = table.getStorage().getStorageFormat();
+            Optional<HiveStorageFormat> hiveStorageFormat = getHiveStorageFormat(storageFormat);
```

Comment:
What does it mean to get an empty hiveStorageFormat ? Shouldn't we throw in that case as well.

Reply:
I thought it could be empty during schema evolution but not sure if that applies here. Added a check to throw for empty hiveStorageFormat as well.